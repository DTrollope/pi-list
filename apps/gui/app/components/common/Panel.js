import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from 'components/common/Icon';

const propTypes = {
    className: PropTypes.string,
    title: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired,
    noPadding: PropTypes.bool,
    noBorder: PropTypes.bool,
    fullHeight: PropTypes.bool,
    backgroundColor : PropTypes.oneOf(['red','green', 'yellow', 'blue', 'white', "gray"])
};

const defaultProps = {
    className: '',
    title: '',
    noPadding: false,
    noBorder: false,
    fullHeight: false,
};

const Panel = (props) => {
    const cardClassNames = classNames(
        'lst-panel',
        props.className
    );

    const containerClassNames = classNames(
        'lst-panel__container',
        {
            'lst-panel--no-padding': props.noPadding,
            'lst-panel--padding': !props.noPadding,
            'lst-panel--no-border': props.noBorder,
            'lst-panel--border': !props.noBorder,
            'lst-panel--full-height': props.fullHeight
        }
    );

    return (
        <div className={cardClassNames}>
            <div className={containerClassNames}>
                {props.title && (
                    <div className="lst-panel__header lst-truncate">
                        <h2>{props.icon && <Icon value={props.icon}/>}{props.title}</h2>
                    </div>
                )}
                {props.children}
            </div>
        </div>
    );
};

Panel.propTypes = propTypes;
Panel.defaultProps = defaultProps;

export default Panel;
