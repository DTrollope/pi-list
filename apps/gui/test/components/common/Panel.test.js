import React from 'react';
import { shallow } from 'enzyme';
import Panel from 'components/common/Panel';
import renderer from 'react-test-renderer';

describe('components/common/Panel', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Panel>
                Just a panel!
            </Panel>
        );
    });

    describe('Panel Component', () => {
        it('should include a custom className', () => {
            wrapper.setProps({ className: 'custom-classname' });

            expect(wrapper.find('.lst-panel').hasClass('custom-classname')).toBe(true);
        });

        it('should render the correct children', () => {
            expect(wrapper.find('.lst-panel').text()).toBe('Just a panel!');
        });
    });

    describe('Panel Snapshots', () => {
        it('should render a panel with padding', () => {
            const tree = renderer
                .create(<Panel>Just a panel!</Panel>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a panel with no padding', () => {
            const tree = renderer
                .create(<Panel noPadding>Just a panel!</Panel>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a panel with header', () => {
            const tree = renderer
                .create(<Panel title="test title" noPadding>Just a panel!</Panel>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    /*describe('Loader Snapshots', () => {
        it('should render a default loader', () => {
            const tree = renderer
                .create(<Loader />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a mini loader', () => {
            const tree = renderer
                .create(<Loader size="mini"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a small loader', () => {
            const tree = renderer
                .create(<Loader size="small"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a large loader', () => {
            const tree = renderer
                .create(<Loader size="large"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
    */
});
