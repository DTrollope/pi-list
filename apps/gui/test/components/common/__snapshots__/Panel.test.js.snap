// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Panel Panel Snapshots should render a panel with header 1`] = `
<div
  className="lst-panel"
>
  <div
    className="lst-panel__container lst-panel--no-padding"
  >
    <div
      className="lst-panel__header lst-truncate"
    >
      <h2>
        test title
      </h2>
    </div>
    Just a panel!
  </div>
</div>
`;

exports[`components/common/Panel Panel Snapshots should render a panel with no padding 1`] = `
<div
  className="lst-panel"
>
  <div
    className="lst-panel__container lst-panel--no-padding"
  >
    
    Just a panel!
  </div>
</div>
`;

exports[`components/common/Panel Panel Snapshots should render a panel with padding 1`] = `
<div
  className="lst-panel"
>
  <div
    className="lst-panel__container lst-panel--padding"
  >
    
    Just a panel!
  </div>
</div>
`;
